datadog/api/aws_integration.py

```diff
@@ -0,0 +1,222 @@
+        excluded by adding ! before the tag. e.x. \
+        env:production,instance-type:c1.*,!region:us-east-1 For more information \
+        on EC2 tagging, see the AWS tagging documentation.
+        :type filter_tags: list of strings
+
+        :param host_tags: Array of tags (in the form key:value) to add to all hosts and \
+        metrics reporting through this integration.
+        :type host_tags: list of strings
+
+        :param account_specific_namespace_rules: An object (in the form \
+        {"namespace1":true/false, "namespace2":true/false}) that enables \
+        or disables metric collection for specific AWS namespaces for this \
+        AWS account only. A list of namespaces can be found at the \
+        /v1/integration/aws/available_namespace_rules endpoint.
+        :type account_specific_namespace_rules: dictionary
+
+        :returns: Dictionary representing the API's JSON response
+
+        The following will depend on whether role delegation or access keys are being used.
+        If using role delegation, use the fields for role_name and account_id.
+        For access keys, use fields for access_key_id and secret_access_key.
+
+        Both the existing fields and new fields are required no matter what. i.e. If the config is \
+        account_id/role_name based, then `account_id`, `role_name`, `new_account_id`, and \
+        `new_role_name` are all required.
+
+        For access_key based accounts, `access_key_id`, `secret_access_key`, `new_access_key_id`, \
+        and `new_secret_access_key` are all required.
+
+        >>> account_id = "<EXISTING_AWS_ACCOUNT_ID>"
+        >>> role_name = "<EXISTING_AWS_ROLE_NAME>"
+        >>> access_key_id = "<EXISTING_AWS_ACCESS_KEY_ID>"
+        >>> secret_access_key = "<EXISTING_AWS_SECRET_ACCESS_KEY>"
+        >>> new_account_id = "<NEW_AWS_ACCOUNT_ID>"
+        >>> new_role_name = "<NEW_AWS_ROLE_NAME>"
+        >>> new_access_key_id = "<NEW_AWS_ACCESS_KEY_ID>"
+        >>> new_secret_access_key = "<NEW_AWS_SECRET_ACCESS_KEY_ID>"
+        >>> filter_tags = ["<KEY>:<VALUE>"]
+        >>> host_tags = ["<KEY>:<VALUE>"]
+        >>> account_specific_namespace_rules = {"namespace1":true/false, "namespace2":true/false}
+
+        >>> api.AwsIntegration.update(account_id=account_id, role_name=role_name, \
+        new_account_id=new_account_id, new_role_name=new_role_name, \
+        filter_tags=filter_tags,host_tags=host_tags,\
+        account_specific_namespace_rules=account_specific_namespace_rules)
+        """
+        params = {}
+        if body.get('account_id') and body.get('role_name'):
+            params['account_id'] = body.pop('account_id')
+            params['role_name'] = body.pop('role_name')
+            if body.get('new_account_id'):
+                body['account_id'] = body.pop('new_account_id')
+            if body.get('new_role_name'):
+                body['role_name'] = body.pop('new_role_name')
```

Comment:
If I read this correctly, there a few cases where I'm not sure what will happen. If body is:

```python
body = {"account_id": "foobar", "new_role_name": "foobaz"}
```
Then it'll be pass as is to the API. Is that expected?
(I don't know the API :))

Reply:
In this case: 

```
from datadog import initialize, api

options = {
    'api_key': '<DD_API_KEY>',
    'app_key': '<DD_APP_KEY>'
}

initialize(**options)

print(api.AwsIntegration.update(
    account_id="<Existing_account_id>",
    new_role_name="Testing"
))
```

Would return an error from the API:
`{u'errors': [u"Additional properties are not allowed (u'new_role_name' was unexpected)"]}`

This is expected as well.

Reply:
Cool, as long as that's what we want, fine with me :)

Reply:
Thanks @jd !